ambry-server/src/main/java/com/github/ambry/server/AmbryServer.java

```diff
@@ -263,16 +255,44 @@ public void startup() throws InstantiationException {
          * having a separate CloudtoStorageManager for cloud offers the flexibility to override methods and add more
          * suited to cloud.
          */
-        networkClientFactory = new RecoveryNetworkClientFactory(properties, registry, clusterMap, storageManager,
-            accountService);
-        vcrClusterSpectator = null; // Server does not talk to vcr during recovery
-        _recoveryManager =
-            new RecoveryManager(replicationConfig, clusterMapConfig, storeConfig, storageManager,
-                storeKeyFactory, clusterMap, scheduler, nodeId, networkClientFactory, registry, notificationSystem,
-                storeKeyConverterFactory, serverConfig.serverMessageTransformer, null,
-                clusterParticipants.get(0));
-        _recoveryManager.start();
-      } else {
+        networkClientFactory =
+            new RecoveryNetworkClientFactory(properties, registry, clusterMap, storageManager, accountService);
+        recoveryManager =
+            new RecoveryManager(replicationConfig, clusterMapConfig, storeConfig, storageManager, storeKeyFactory,
+                clusterMap, scheduler, nodeId, networkClientFactory, registry, notificationSystem,
+                storeKeyConverterFactory, serverConfig.serverMessageTransformer, null, null);
+        recoveryManager.start();
+      } else if (serverConfig.serverExecutionMode.equals(ServerExecutionMode.DATA_VERIFICATION_MODE.toString())) {
+        // Backup integrity monitor here because vcr does not have code to store to disk and the code to create that is here
+        RecoveryManager recoveryManager =
```

Comment:
Looks like we should try to shutdown this recovery manager when shutting down the server. There is already a class variable recoveryManager, why not just use it?

Reply:
good pt ! used the existing one.